tests/rptest/tests/compaction_recovery_test.py

```diff
@@ -79,21 +102,100 @@ def test_index_recovery(self):
                    backoff_sec=2,
                    err_msg="Timeout waiting for partitions to recover.")
 
-    def produce_until_segments(self, count):
-        partitions = []
-        kafka_tools = KafkaCliTools(self.redpanda)
 
-        def check_partitions():
-            kafka_tools.produce(self.topic, 1024, 1024)
-            storage = self.redpanda.storage()
-            partitions[:] = storage.partitions("kafka", self.topic)
-            return partitions and all(
-                map(lambda p: len(p.segments) > count and p.recovered(),
-                    partitions))
+class CompactionRecoveryUpgradeTest(RedpandaTest):
+    OLD_VERSION = (22, 2, 8)
```

Comment:
I think latest redpanda_installer changes might have included the ability to just say `(22, 2)` and get the latest version from that series

Reply:
But then if we backport and release this changes in a 22.2 release, the test will stop working, right?

Reply:
That's true, but I don't think we will backport it to 22.2 -- format changes like this are for major releases.